### PR TITLE
Fix E2E test cases and setup for GHA

### PR DESCRIPTION
**Pull request by mvitale1989**
#### Summary

This PR tackles the test failures seen in the following GHA-based E2E run: https://automation-dashboard.vercel.app/cycles/1d2ed8a4-410d-413f-9b3a-46830b96090b

Here is a summary of the underlying issues:
- Compared to cypress-ui-automations, we initiate the tests _very quickly_ after the server starts. This may happen before migrations have been completed (e.g. `MM-T1537`, `MM-T2871`, `MM-23018`). We solve this by tracking in the `Systems` table whether the migration(s) we need have been completed or not.
- Some tests expected the tour tips to be there (e.g. `MM-T4429_2`). Others are there to explicitly run the tour (e.g. `tests/integration/channels/collapsed_reply_threads/crt_tour_spec.js`)
- The `TeamSettings.EnableOpenServer` setting can change during tests, so it cannot be set through an env var (a problem for e.g. `MM-47222`)
- Actual test fixes:
  * `tests/integration/channels/files_and_attachments/image_link_preview_spec.js` (`MM-T2389`)
  * `integration/channels/markdown/markdown_image_spec.js`
  * `integration/channels/settings/display/theme/code_theme_colors_spec.js`
  * `integration/channels/team_settings/join_closed_team_with_not_allowed_email_spec.js`

Some tests need to be checked again, as they succeeded on my local machine after I applied fixes for the above:
- tests/integration/channels/enterprise/system_console/channel_members_spec.js
- tests/integration/channels/enterprise/system_console/channel_moderation/channel_mentions_spec.js
- integration/channels/system_console/site_configuration/link_customization_e20_2_spec.js

Two tests in particular deserve a deeper look:
- `integration/channels/toast/unread_with_bottom_start_toast_spec.js`: without the `wait(50)`, the number of messages reported by the toast may be lower than the actually posted messages. We need to verify if this is an actual issue.
- `integration/playbooks/channels/slash_command/test_spec.js`: the Automation Dashboard shows the individual test cases as successful, but then gets the number wrong in the file summary. You can see this in the above linked run already.

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-7076

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: e2e-tests/.ci/server.generate.sh [hunk 1]
```diff
@@ -59,13 +58,10 @@ services:
       MM_PLUGINSETTINGS_ENABLED: "true"
       MM_PLUGINSETTINGS_ENABLEUPLOADS: "true"
       MM_PLUGINSETTINGS_AUTOMATICPREPACKAGEDPLUGINS: "true"
```
Comment (saturninoabril) [inline e2e-tests/.ci/server.generate.sh:60]: These 3 should be modified during setup too and not as env variable for there are tests that modify their values.

File: e2e-tests/cypress/tests/integration/channels/files_and_attachments/image_link_preview_spec.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ describe('Image Link Preview', () => {
 
         const markdownImageText = 'exampleImage';
         const markdownImageSrc = 'https://docs.mattermost.com/_images/icon-76x76.png';
-        const markdownImageSrcEncoded = encodeURIComponent(markdownImageSrc); // Since the url preview will be encoded string
+        const markdownImageSrcEncoded = encodeURIComponent(`/api/v4/image?url=${encodeURIComponent(markdownImageSrc)}`); // Since the url preview will be an encoded string
```
Comment (saturninoabril) [inline e2e-tests/cypress/tests/integration/channels/files_and_attachments/image_link_preview_spec.js:105]: ❤️

**Review by saturninoabril — APPROVED** [review]
Left one request but all good to me, and so happy to see all the fixes! Thanks @mvitale1989! Approving now.

**Comment by mvitale1989** [comment]
I solved a few additional issues, notably:
- The `Billing History` view has been removed from selfhosted in this PR: https://github.com/mattermost/mattermost/pull/26853/files#diff-13d21cc3b35729521a83fcd5a4c8518d05e09ca6e802ca1a647e4c65c26f6df6L166
  * cross-check: its merging time matches the time between the test case's [last success](https://automation-dashboard.vercel.app/cycles/8712f5e1-dcbb-431e-9d82-d901292b913d) and its [first failure](https://automation-dashboard.vercel.app/cycles/b6f8a403-f387-426d-8a2b-97c26d3a6aae) in the daily tests.
  * Also solved a known issue for the `Mattermost Users` test case
- `channel_header_spec.js`: Cypress doesn't seem to be able to hover the textbox. But we can still look for its existence and click it.
- `sidebar_link_navigation_e20_spec.js`
- `license_no_license_spec.js`

There are a few more to solve, and I also want to understand what's happening with `image_link_preview_spec.js`: it is solved for GHA-based tests, but I'm not sure it is the right approach, as gitlab-based tests were suceeding all along. I have asked about it here: https://hub.mattermost.com/private-core/pl/ra1s64irufgz8x6qj787z7mony

**Review by saturninoabril — APPROVED** [review]
Awesome, thanks @mvitale1989!

**Review by yasserfaraazkhan — APPROVED** [review]
🎉

**Pull request merged**